### PR TITLE
fix: engines not starting properly in msfs 2024

### LINKED ISSUE #9575: [A32NX] FS2024: Engines Fail to Start
### Aircraft Version

Stable

### Build info

```json
{
    "built": "2024-11-21T09:20:01+00:00",
    "ref": "refs/heads/master",
    "sha": "1d42acddaf69e496f74e1ca1d6967183eaecf65c",
    "actor": "tracernz",
    "event_name": "manual",
    "pretty_release_name": "master:1d42acdd",
    "version": "a32nx-v0.13.0-dev.1d42acd"
}
```

### Describe the bug

- N2% take an unexpectedly long time to spool up.
- During the start phase, the engines stop by themselves and returns both N1 & N2 values to 0.
- EGT Values remain elevated when ignition & Engine 1 Switch stays on.

### Expected behavior

- Engines should start normally with expected performance.

### Steps to reproduce

1. Perform the standard engine start procedure as outlined in the official FlyByWire (FBW) documentation.
2. Observe the engine behavior during the startup phase.

### References (optional)

_No response_

### Additional info (optional)

Do note that this bug is only found in FS2024.

### Discord Username (optional)

MJ

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #9575

## Summary of Changes

execute_calculator_code does not seem to work properly in this case (and is discouraged anyway). Replaced with `trigger_key_event` (deprecated but working in both sims), which brings the correct behaviour back again.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Spawn C&D in both sims ideally and perform engine starts with APU and ASU. N2 should rise quickly after engine master switches have been turned on

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

**Review by tracernz — APPROVED** [review]

**Comment by tracernz** [comment]
No changelog needed as there's no user-visible change.

**Comment by flogross89** [comment]
Does this only apply to Rust code or also to C++ code? ``execute_calculator_code`` is used extensively there

**Comment by Saschl** [comment]
> Does this only apply to Rust code or also to C++ code? `execute_calculator_code` is used extensively there

Will need to check that as well 👍 Most of the calls do not seem to take parameters which seems to work fine, but I haven't checked very thoroughly yet

File: fbw-common/src/wasm/systems/systems_wasm/src/electrical.rs [hunk 1]
```diff
@@ -83,28 +88,27 @@ pub(super) fn auxiliary_power_unit(
...
-    execute_calculator_code::<()>("1 (>K:APU_OFF_SWITCH, Number)");
+    trigger_key_event(KEY_APU_OFF_SWITCH, 1);
 }
 
 fn supply_bleed(on: bool) {
-    let action = if on { "1" } else { "0" };
-    execute_calculator_code::<()>(&format!("{} (>K:APU_BLEED_AIR_SOURCE_SET, Bool)", action));
+    let action: u32 = if on { 1 } else { 0 };
+
+    trigger_key_event(KEY_APU_BLEED_AIR_SOURCE_SET, action);
```
Comment (Gurgel100) [inline fbw-common/src/wasm/systems/systems_wasm/src/electrical.rs:113]:
```suggestion
    trigger_key_event(KEY_APU_BLEED_AIR_SOURCE_SET, on.into());
```

**Review by Gurgel100 — CHANGES_REQUESTED** [review]

File: fbw-common/src/wasm/systems/systems_wasm/src/electrical.rs (hunk 1 above)
Comment (Saschl) [reply fbw-common/src/wasm/systems/systems_wasm/src/electrical.rs]: Ohh nice, fixed :)

**Review by Gurgel100 — APPROVED** [review]

**Comment by denmax18** [comment]
We have problems with the simbrief tool open all power disconnecting. A388

**Pull request merged**